pkg/cosign/rego/rego.go

```diff
@@ -33,23 +34,31 @@ func ValidateJSON(jsonBody []byte, entrypoints []string) error {
 
 	query, err := r.PrepareForEval(ctx)
 	if err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	var input interface{}
 	dec := json.NewDecoder(bytes.NewBuffer(jsonBody))
 	dec.UseNumber()
 	if err := dec.Decode(&input); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	rs, err := query.Eval(ctx, rego.EvalInput(input))
 	if err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if rs.Allowed() {
-		return nil
+		return errs
 	}
-	return errors.New("rego validation failed")
+
+	for _, result := range rs {
+		for _, expression := range result.Expressions {
+			for _, v := range expression.Value.([]interface{}) {
+				errs = append(errs, errors.Errorf("%s", v.(string)))
```

Comment:
We could be more cautious here:

```rego
deny[{}] = { true }
```
would make this panic

Reply:
what do you suggest I do? 🙏

Reply:
something along those lines, maybe?
```go
if s, ok := v.(string); ok {
	errs = append(errs, fmt.Errorf(s))
} else {
	errs = append(errs, fmt.Errorf("%s": v))
}
```